packages/pigeon/lib/gobject_generator.dart

```diff
@@ -0,0 +1,2071 @@
+          asyncArgs.add('size_t ${paramName}_length');
+        }
+      }
+      asyncArgs.addAll(<String>[
+        'GCancellable* cancellable',
+        'GAsyncReadyCallback callback',
+        'gpointer user_data'
+      ]);
+      indent.newln();
+      indent.writeln('/**');
+      indent.writeln(' * ${methodPrefix}_$methodName:');
+      indent.writeln(' * @api: a #$className.');
+      for (final Parameter param in method.parameters) {
+        final String paramName = _snakeCaseFromCamelCase(param.name);
+        indent.writeln(
+            ' * @$paramName: ${param.type.isNullable ? '(allow-none): ' : ''}parameter for this method.');
+        if (_isNumericListType(param.type)) {
+          indent.writeln(' * @${paramName}_length: length of $paramName.');
+        }
+      }
+      indent
+          .writeln(' * @cancellable: (allow-none): a #GCancellable or %NULL.');
+      indent.writeln(
+          ' * @callback: (scope async): (allow-none): a #GAsyncReadyCallback to call when the call is complete or %NULL to ignore the response.');
+      indent
+          .writeln(' * @user_data: (closure): user data to pass to @callback.');
+      indent.writeln(' *');
+      addDocumentationComments(
+          indent, method.documentationComments, _docCommentSpec);
+      indent.writeln(' */');
+      indent.writeln(
+          "void ${methodPrefix}_$methodName(${asyncArgs.join(', ')});");
+
+      final List<String> finishArgs = <String>[
+        '$className* api',
+        'GAsyncResult* result',
+        'GError** error'
+      ];
+      indent.newln();
+      indent.writeln('/**');
+      indent.writeln(' * ${methodPrefix}_${methodName}_finish:');
+      indent.writeln(' * @api: a #$className.');
+      indent.writeln(' * @result: a #GAsyncResult.');
+      indent.writeln(
+          ' * @error: (allow-none): #GError location to store the error occurring, or %NULL to ignore.');
+      indent.writeln(' *');
+      indent.writeln(' * Completes a ${methodPrefix}_$methodName() call.');
+      indent.writeln(' *');
+      indent.writeln(' * Returns: a #$responseClassName or %NULL on error.');
+      indent.writeln(' */');
+      indent.writeln(
+          "$responseClassName* ${methodPrefix}_${methodName}_finish(${finishArgs.join(', ')});");
+    }
+  }
+
+  // Write the API response classes.
+  void _writeFlutterApiRespondClass(
+      Indent indent, String module, Api api, Method method) {
+    final String responseName = _getResponseName(api.name, method.name);
+    final String responseClassName = _getClassName(module, responseName);
+    final String responseMethodPrefix = _getMethodPrefix(module, responseName);
+    final String primitiveType =
+        _getType(module, method.returnType, primitive: true);
+
+    indent.newln();
+    _writeDeclareFinalType(indent, module, responseName);
+
+    indent.newln();
+    indent.writeln('/**');
+    indent.writeln(' * ${responseMethodPrefix}_is_error:');
```

Comment:
It looks like we just need to have these five methods for Host API response classes too.